### PR TITLE
Add short_text management + example for time and net block

### PR DESCRIPTION
**Pull request by matclab**
This PR propose to add short_text management as a response to #676. 

The features are:
- a widget may set full_text and short_text which are then automatically used by i3bar depending upon the available space in the bar (this is the default behavior) — this is similar to PR #693 ,
- the user may right or middle click on the block to switch from the default behavior to a widget where both full_text and shot_text are set to either the wide text or the narrow text.

This allow the user to enforce a widget view (full, short or default) by clicking or by configuration.

Does this behavior suites you ?

NB. I'm a rust and i3 beginner, so, do not hesitate to review and suggest improvement.

### CODE REVIEW

File: src/blocks/time.rs [hunk 1]
```diff
@@ -78,44 +97,59 @@ impl ConfigBlock for Time {
...
     }
 }
 
 impl Block for Time {
     fn update(&mut self) -> Result<Option<Update>> {
-        let time = match self.timezone {
-            Some(tz) => Utc::now().with_timezone(&tz).format(&self.format),
-            None => Local::now().format(&self.format),
+        self.time = match self.timezone {
+            Some(tz) => Local.from_utc_datetime(&Utc::now().with_timezone(&tz).naive_local()),
```
Comment (ammgws) [inline src/blocks/time.rs:119]: Unrelated change? If so please open a new PR.

**Comment by ammgws** [comment]
Middle click might be the best bet for this as no block uses it:
https://github.com/greshake/i3status-rust/issues/697#issuecomment-632134634

File: src/blocks/time.rs (hunk 1 above)
Comment (matclab) [reply src/blocks/time.rs:119]: Oups, it was a leftover of a first try. I'll push a commit that remove self.time. However I need to store the datetime instead of the formated string as the datetime is then formatted differently for short_text and full_text

File: src/blocks/time.rs [hunk 2]
```diff
@@ -78,44 +97,59 @@ impl ConfigBlock for Time {
...
+                match e.button {
+                    MouseButton::Left => {
+                        if let Some(ref cmd) = self.on_click {
+                            spawn_child_async("sh", &["-c", cmd])
+                                .block_error("time", "could not spawn child")?;
+                        }
+                    }
+                    MouseButton::Right | MouseButton::Middle => {
+                        self.width = self.width.next();
+                        let _ = self.update();
```
Comment (matclab) [inline src/blocks/time.rs:142]: @ammgws Is it better to do the update via a `Sender` like it is done in custom block? 
```
            self.tx_update_request.send(Task {
                id: self.id.clone(),
                update_time: Instant::now(),
            })?;
```

File: src/widgets/button.rs [hunk 3]
```diff
@@ -60,7 +81,16 @@ impl ButtonWidget {
     }
 
     pub fn set_text<S: Into<String>>(&mut self, content: S) {
-        self.content = Some(content.into());
+        let content = Some(content.into());
```
Comment (matclab) [inline src/widgets/button.rs:84]: I guess that this function should probably always set both `content` and `short_content` to keep existing blocks working as they do now (else, we may have blocks that get displayed with the `short_text` being the `full_text` that was given at initialisation instead of the actual `full_text`)

**Comment by ammgws** [comment]
Sorry I am not sure if I can review this soon. Have you been using it so far?

**Comment by matclab** [comment]
Hello,

There's no hurry.

Yes I use it daily. It works quite well but the three state clicking
behaviour is a bit frustrating (the user doesn't know on which state is the
block, so the user shall sometimes click twice to see a change). For now I
don't see how I can do better and I think it is better than having some
blocks not displayed at all.

I'm also a bit frustrated by i3bar behaviour which is : full_text for every
blocks or short_text for every blocks. I would have prefer that it used
short_text only for the minimum of needed blocks.

Anyway I think the PR need some documentation and perhaps I can also port
the behaviour to other blocks.

**Comment by ammgws** [comment]
> I'm also a bit frustrated by i3bar behaviour which is : full_text for every
> blocks or short_text for every blocks. I would have prefer that it used
> short_text only for the minimum of needed blocks.

It might be worth looking at the i3 source to see how it is handling this.
It might be able to be fixed trivially.

**Comment by matclab** [comment]
Some progress information : I'm still working on this, but struggling a little with GraphWidget not displaying shorttext. Some more investigations needed.

**Comment by ammgws** [comment]
I opened a feature request over at i3: https://github.com/i3/i3/issues/4113

Feel free to add to the (lack of) discussion over there since it might lead to something!

**Comment by ammgws** [comment]
Looks like our feature request was accepted by i3

**Comment by matclab** [comment]
close in favor of #1207
